### DISCUSSION TITLE
General questions about flow control and hydrogym

### DISCUSSION THREAD

**Discussion by ReHoss** [dynamicslab/hydrogym#93] [Q&A]
Hi,

I have the following questions regarding both Fluid Control in general and its practical implementation (my questions might seem obvious to you but I don't have a mechanical engineering background).

- What do you call a natural flow and why the initial condition $x_0$ is often omited ?

If I understand correctly, we are not really interested in transient behavior in CFD control but more on the _stationary_ part of the dynamical system, so-called _natural flow_ (please correct me if necessary). Then, this would be the reason initial conditions $x_0$ are rarely mentioned in the landmark references @jcallaham collected https://github.com/dynamicslab/hydrogym/discussions/72.

Hence, I could imagine that transient dynamics are too short temporally to be effectively controlled nowadays and focusing on controlling from the **attractor** is enough ?

- In Reinforcement Learning, we are often limited to episodic control (i.e. controlled trajectories with finite time-horizon) starting from an initial state $\mathbb{P}_{X_0}$. In this case, what is the recommended or more realistic or common way to choose the distribution of the initial flow state ?

- More generally are there any references on how and when a control is applied ? Episodic training procedure methodology ?
About that part, I will look at Brunton's or J. Rabault's papers.

- How to apply control having larger time-resolution $\delta_t^{control}$ than the solver time resolution $\delta_t^{IPCS}$ ?
In general one want to distinguish the frequency the control is applied from the frequency on which the continuous _Navier-Stokes_ operator is discretised.

**Comment by jcallaham** [comment]
So for your first question, yes, the "post-transient flow" is basically the setting we've mostly been thinking about here, although there are certainly situations where controlling the transient is important.  Just depends on the application.  But yes, that's the reason the initial condition isn't discussed much.  Also, we haven't settled on a good way of sampling initial conditions from a distribution in the package yet, so that's still left up to the user.  For instance, you could do something like run a post-transient simulation with no forcing and save a number of snapshots, then randomly select from these as an initialization.

I'm not sure I understand the last two questions.  What do you mean by "how and when a control is applied?"  Training procedure in general is beyond the scope of this package - we're just trying to provide environments, agnostic to the type of control algorithms.  Similarly, I'm not sure what you mean by having different time steps for control and CFD solver.  They need to be commensurate, at least.  Do you mean integrating for multiple steps of the CFD solver before updating the control input?  We could probably support that without too much trouble, or you could implement it yourself by adding a kwarg to `FlowEnv`.  The other way around ($\delta_t^u < \delta_t^x$) I think doesn't really make sense.
